docs/extend/assets.md

```diff
@@ -0,0 +1,8 @@
+# Extension Assets
+
+Some extensions might want to include assets like images or JSON files in their source code (note that this is not the same as uploads, which would probably require a [filesystem disk](filesystem.md)).
+
+This is actually very easy to do. Just create an `assets` folder at the root of your extension, and place any asset files there.
+Flarum will then automatically copy those files to its own `assets` directory every time the extension is enabled or [`php flarum assets:publish`](../console.md) is executed.
+
+Those files are copied into Flarum's assets folder and the URL to these files can be retrieved like this from the javascript: `app.forum.attribute('baseUrl') + '/assets/extensions/EXTENSION_ID/pumpkin.jpg'`, where EXTENSION_ID is the internal name of the extension (based on the package name, without / or flarum-ext-) and pumpkin.jpg is an example filename.
```

Comment:
Love the filename ahah. I wonder where this was copied from :stuck_out_tongue: 

We don't have a way to access the assets based on the actual disk, do we? If a user changes the assets disk as described in another file above, then this will break. Not sure if all of Flarum is impacted (logo, avatars, ... do they break or automatically switch if you change the disk public URL?)

Reply:
Just rerun `php flarum assets:publish`

Reply:
I was referring to `app.forum.attribute('baseUrl') + '/assets/extensions/EXTENSION_ID/pumpkin.jpg'` likely not working in case the assets folder is customized.

I see that for logo and favicon we use `$assetsFilesystem->url($assetPath)` in the backend and serialize it to the frontend https://github.com/flarum/core/blob/d64750b3eb7de28d82727105b69c213cae5befe7/src/Api/Serializer/ForumSerializer.php#L115-L135 this is probably what we should also suggest to extensions?

I know I'm the one who originally shared the snippet that uses `baseUrl`, but if we include that in the documentation it's important this will only work with the local assets filesystem.

Reply:
Oh, this is true... Yeah I'll update that, good point!